### ISSUE TITLE
Python 3.10.0a6 test failure

### ISSUE THREAD

**Issue by jamesjer** [aleaxit/gmpy#296]
The Fedora project is experimenting with python 3.10.  The most recent alpha release, version 3.10.0a6, includes this change (see https://docs.python.org/3.10/whatsnew/3.10.html):
```
If object.__ipow__() returns NotImplemented, the operator will correctly fall back to object.__pow__() and object.__rpow__() as expected. https://bugs.python.org/issue38302
```

This leads to two gmpy2 2.1.0b5 test failures:
```
**********************************************************************
File "/builddir/build/BUILD/gmpy-gmpy2-2.1.0b5/test/test_gmpy2_mpz_inplace.txt", line 150, in test_gmpy2_mpz_inplace.txt
Failed example:
    x **= -2
Differences (ndiff with -expected +actual):
      Traceback (most recent call last):
    -   File "<stdin>", line 1, in <module>
    - TypeError: unsupported operand type(s) for ** or pow(): 'mpz' and 'int'
    +   File "/usr/lib64/python3.10/doctest.py", line 1340, in __run
    +     exec(compile(example.source, filename, "single",
    +   File "<doctest test_gmpy2_mpz_inplace.txt[57]>", line 1, in <module>
    +     x **= -2
    + ValueError: pow() exponent cannot be negative
**********************************************************************
File "/builddir/build/BUILD/gmpy-gmpy2-2.1.0b5/test/test_gmpy2_mpz_inplace.txt", line 156, in test_gmpy2_mpz_inplace.txt
Failed example:
    x **= mpfr(2)
Differences (ndiff with -expected +actual):
    - Traceback (most recent call last):
    -   File "<stdin>", line 1, in <module>
    - TypeError: unsupported operand type(s) for ** or pow(): 'mpz' and 'mpfr'
**********************************************************************
1 items had failures:
   2 of  85 in test_gmpy2_mpz_inplace.txt
***Test Failed*** 2 failures.
```

Both are expected to raise TypeErrors.  The first now raises a ValueError, and the second succeeds, returning an mpfr result.

**Comment by jamesjer** [comment]
To work around the test failures for now, we are going to apply this patch if and only if python 3.10 is used to build:
https://src.fedoraproject.org/rpms/python-gmpy2/blob/rawhide/f/python-gmpy2-pow.patch

We are happy to replace that with whatever better solution you come up with.

**Comment by casevh** [comment]
This in interesting. Yes, there is a change due to a bug fix in 3.10. But the more significant issue is that mpz(7)**-2 raises an exception but 7**-2 returns a float. I will make gmpy2 return the same results as Python.

**Comment by casevh** [comment]
Fixes have been committed. Thanks for the report.

**Comment by skirpichev** [comment]
@casevh, probably this was fixed by db7ce2e and can be closed?

**Comment by casevh** [comment]
Closing. Fix has been done.
